_includes/hero-body.html

```diff
@@ -51,20 +78,71 @@ <h2 class="is-hidden-tablet">RISM Catalog of Musical Sources</h2>
 
 {% if page.permalink == "/index.html" %}
 <script>
+    // update the checkbox when the page loads
+    window.onload = function() {
+        update_checkboxes();
+    }
+
+    // listen to 'Enter' on the input box
     document.getElementById("opac-search-term").addEventListener("keyup", function(event) {
         // Number 13 is the "Enter" key on the keyboard
         if (event.keyCode === 13) {
             // Cancel the default action, if needed
             event.preventDefault();
             // Trigger the button element with a click
-            opac();
+            search();
         }
-    }); 
+    });
+
+    // look at the 'search' local storage value
+    // Default search is the opac
+    function update_checkboxes() {
+        var search = localStorage.getItem("search");
+        if (search == "rism-online") {
+            document.getElementById("rism-online-unchecked").style.display = "none";
+            document.getElementById("rism-online-checked").style.display = "block"; 
+            document.getElementById("opac-unchecked").style.display = "block";
+            document.getElementById("opac-checked").style.display = "none"; 
+        }
+        else {
+            document.getElementById("rism-online-unchecked").style.display = "block";
+            document.getElementById("rism-online-checked").style.display = "none"; 
+            document.getElementById("opac-unchecked").style.display = "none";
+            document.getElementById("opac-checked").style.display = "block"; 
+        }
+    }
+
+    // search from the input box - use selected engine
+    function search() {
+        // Retrieve from the local storage - default is the opac
+        var search = localStorage.getItem("search");
+        if (search == "rism-online") rism_online();
+        else opac();
+    }
 
     function opac() {
+        localStorage.setItem("search", "opac");
```

Comment:
`setItem` can raise an exception if a browser cannot write to localstorage, as is the case in "private browsing" mode.